### ISSUE TITLE
Should `harden-pyyaml` respect use of `FullLoader`?

### ISSUE THREAD

**Issue by drdavella** [pixee/codemodder-python#266]
I've noticed a few changes recently where code is explicitly using `loader=yaml.FullLoader` in `yaml.load`.

What I'm wondering is whether we should respect such explicit usage of `FullLoader` and leave it as-is? The current behavior of the codemod is to replace this with `SafeLoader`, which is obviously safer. However it seems to me that there are potentially valid uses of `FullLoader` as well, and if it is being used explicitly, we should respect it.

The PyYaml [documentation](https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation) has this to say:
> Loads the full YAML language. Avoids arbitrary code execution. This is currently (PyYAML 5.1+) the default loader called by yaml.load(input) (after issuing the warning).

It also says:
> WARNING: As of pyyaml-5.3.1 there are still trivial exploits. Do not use this on untrusted data for now.

So clearly this shouldn't be used on completely untrusted data. However it seems like there are also valid use cases. `FullLoader` is still much more restrictive than `UnsafeLoader` in terms of preventing arbitrary code execution.

TL;DR: I am suggesting that we may want to consider leaving `FullLoader` as-is when it is explicitly used.

**Comment by clavedeluna** [comment]
I would vote to keep the codemod as is and in cases when users truly want to use it:
1. they can tell pixeebot this is what they want via closing PR or some other interaction
2. they can exclude it by line
3. if there is a flake8/pylint code for this, we could add support for disable by linter code

**Comment by drdavella** [comment]
Closing this for now since maybe this isn't too compelling.
